ccp/src/prover.rs

```diff
@@ -127,6 +136,83 @@ impl CCProver {
         // stop background thread
         self.utility_thread.stop().await.map_err(Into::into)
     }
+
+    pub async fn from_saved_state(
```

Comment:
Can we decouple the mutual part with `new` into a separate function and call it from these two?

Reply:
It was my initial idea, but I failed to set idx counter inside the `NewProofHandler` that is very hermetic (and owned by a closure).